src/wgpu_engine.rs

```diff
@@ -539,10 +538,9 @@ impl WgpuEngine {
                                 .with_parent(Some(&query));
                             cpass.set_pipeline(&wgpu_shader.pipeline);
                             cpass.set_bind_group(0, &bind_group, &[]);
-                            let buf = self
-                                .bind_map
-                                .get_gpu_buf(proxy.id)
-                                .ok_or("buffer for indirect dispatch not in map")?;
+                            let buf = self.bind_map.get_gpu_buf(proxy.id).ok_or(
+                                VelloError::UnavailableBufferUsed(proxy.name, "indirect dispatch"),
```

Comment:
I made `"indirect dispatch"` and `"download"` strings here. Should I make an enum for those two so it is easier to match on the error?

Reply:
I think it's fine, personally.